File: backend/src/main/java/com/github/jonashonecker/backend/ticket/domain/Ticket.java
package com.github.jonashonecker.backend.ticket.domain;

import com.github.jonashonecker.backend.user.domain.TicketScoutUser;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("tickets")
public record Ticket(
        String id,
        String projectName,
        String title,
        String description,
        Status status,
        TicketScoutUser author
) {
}

File: frontend/src/types/Ticket.ts
import { User } from "./User.ts";

export type TicketStatus = "OPEN" | "IN_PROGRESS" | "REJECTED" | "CLOSED";

export type Ticket = {
  id: string;
  projectName: string;
  title: string;
  description: string;
  status: TicketStatus;
  author: User;
};

export type TicketRequestDTO = {
  title: string;
  description: string;
};
